include/swift/ClangImporter/ClangImporter.h

```diff
@@ -89,13 +113,15 @@ class ClangImporter final : public ClangModuleLoader {
   ///
   /// \param tracker The object tracking files this compilation depends on.
   ///
+  /// \param dwarfImporterDelegate A helper object that can synthesize
+  /// Clang Decls from debug info. Used by LLDB.
+  ///
   /// \returns a new Clang module importer, or null (with a diagnostic) if
   /// an error occurred.
   static std::unique_ptr<ClangImporter>
-  create(ASTContext &ctx,
-         const ClangImporterOptions &importerOpts,
-         std::string swiftPCHHash = "",
-         DependencyTracker *tracker = nullptr);
+  create(ASTContext &ctx, const ClangImporterOptions &importerOpts,
+         std::string swiftPCHHash = "", DependencyTracker *tracker = nullptr,
+         std::unique_ptr<DWARFImporterDelegate> dwarfImporterDelegate = {});
```

Comment:
LLDB creates the ClangImporter and hands it over to the swift::ASTContext. Since lldb_private::SwiftASTContext owns the unique_ptr to the swift::ASTContext, I can just make the delegate a proper member of lldb_private::SwiftASTContext and everything will still be destroyed in the correct order.

Reply:
https://github.com/apple/swift/pull/26676 + https://github.com/apple/swift-lldb/pull/1891